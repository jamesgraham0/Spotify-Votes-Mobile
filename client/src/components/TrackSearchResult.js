import { StyleSheet, Text, View, Image } from 'react-native';
import React from 'react';
import { BlurView } from 'expo-blur';

const TrackSearchResult = ({ track, inQueue }) => {

    return (
        <View>
            {inQueue ? 
            <View style={styles.container}>
                <Image 
                    style={styles.img}
                    source={{uri: track.smallImage}}
                />
                <View style={styles.titleArtistContainer}>
                    <Text style={styles.title}>{track.title}</Text>
                    <Text style={styles.artist}>{track.artist}</Text>
                </View>
                <BlurView intensity={30} tint={'dark'} style={styles.containerBlur}>
                    <Text style={{color: 'white', fontSize: 30}}>Added to queue</Text>
                </BlurView>
            </View> :
            <View style={styles.container}>
                <Image 
                    style={styles.img}
                    source={{uri: track.smallImage}}
                />
                <View style={styles.titleArtistContainer}>
                    <Text style={styles.title}>{track.title}</Text>
                    <Text style={styles.artist}>{track.artist}</Text>
                </View>
            </View>
            }       
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        backgroundColor: '#080808',
        marginVertical: 10,
        marginHorizontal: 20,
        padding: 5,
        borderRadius: 5,
        height: 80,
    },
    containerBlur: {
        position: 'absolute',
        justifyContent: 'center',
        alignItems: 'center',
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
        borderRadius: 5,
        backgroundColor: 'rgba(0, 50, 100, 0.2)',
        overflow: 'hidden',
    },
    titleArtistContainer: {
        marginHorizontal: 30,
        marginVertical: 10,
        width: 180,
        alignSelf: 'center',
    },
    title: {
        color: '#fff',
        fontWeight: 'bold',
        fontSize: 20,
    },
    artist: {
        fontSize: 16,
        color: '#fff',
    },
    img: {
        backgroundColor: 'black',
        width: 70,
        height: 70,
        borderRadius: 5,
    }
});

export default TrackSearchResult;