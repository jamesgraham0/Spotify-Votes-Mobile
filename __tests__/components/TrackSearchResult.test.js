import React from 'react';
import renderer from 'react-test-renderer';
import TrackSearchResult from '../../client/src/components/TrackSearchResult';

describe('TrackSearchResult', () => {
    it('should render the TrackSearchResult component', () => {
        const track = {
            title: 'Test Track',
            artist: 'Test Artist',
            smallImage: 'test.jpg',
        };
        const inQueue = false;
        const tree = renderer.create(<TrackSearchResult track={track} inQueue={inQueue}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it ('should render the TrackSearchResult component with blur', () => {
        const track = {
            title: 'Test Track',
            artist: 'Test Artist',
            smallImage: 'test.jpg',
        };
        const inQueue = true;
        const tree = renderer.create(<TrackSearchResult track={track} inQueue={inQueue}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});