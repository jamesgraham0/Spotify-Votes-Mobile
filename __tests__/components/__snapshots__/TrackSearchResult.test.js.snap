// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackSearchResult should render the TrackSearchResult component 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#080808",
        "borderRadius": 5,
        "flex": 1,
        "flexDirection": "row",
        "height": 80,
        "marginHorizontal": 20,
        "marginVertical": 10,
        "padding": 5,
      }
    }
  >
    <Image
      source={
        {
          "uri": "test.jpg",
        }
      }
      style={
        {
          "backgroundColor": "black",
          "borderRadius": 5,
          "height": 70,
          "width": 70,
        }
      }
    />
    <View
      style={
        {
          "alignSelf": "center",
          "marginHorizontal": 30,
          "marginVertical": 10,
          "width": 180,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Test Track
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
          }
        }
      >
        Test Artist
      </Text>
    </View>
  </View>
</View>
`;

exports[`TrackSearchResult should render the TrackSearchResult component with blur 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#080808",
        "borderRadius": 5,
        "flex": 1,
        "flexDirection": "row",
        "height": 80,
        "marginHorizontal": 20,
        "marginVertical": 10,
        "padding": 5,
      }
    }
  >
    <Image
      source={
        {
          "uri": "test.jpg",
        }
      }
      style={
        {
          "backgroundColor": "black",
          "borderRadius": 5,
          "height": 70,
          "width": 70,
        }
      }
    />
    <View
      style={
        {
          "alignSelf": "center",
          "marginHorizontal": 30,
          "marginVertical": 10,
          "width": 180,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Test Track
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
          }
        }
      >
        Test Artist
      </Text>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "transparent",
          },
          {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 50, 100, 0.2)",
            "borderRadius": 5,
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    >
      <ViewManagerAdapter_ExpoBlurView
        blurReductionFactor={4}
        experimentalBlurMethod="none"
        intensity={30}
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        tint="dark"
      />
      <Text
        style={
          {
            "color": "white",
            "fontSize": 30,
          }
        }
      >
        Added to queue
      </Text>
    </View>
  </View>
</View>
`;
